app/models/job_proxy_dispatcher.rb

```diff
@@ -197,10 +230,13 @@ def busy_proxies
     end
   end
 
-  def active_scans
+  def active_scans(target_class = VmOrTemplate)
+    class_name = target_class.base_class.name
     @active_scans ||= begin
       actives = Hash.new(0)
-      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active").where.not(:state => "finished").count
+      actives[@zone] = Job.where(:zone => @zone, :dispatch_status => "active")
```

Comment:
Please add tests for this method to ensure two dispatched jobs (one ContainerImage and one VmOrTemplate) do the right thing when you ask for the number of active scans for one or the other.